Runtime/Scripts/BaseMeshSync.Textures.cs

```diff
@@ -0,0 +1,243 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+
+namespace Unity.MeshSync {
+    partial class BaseMeshSync {
+#if UNITY_EDITOR
+        /// <summary>
+        /// Names of all maps that could be baked to a render texture and might need to be saved to the asset database.
+        /// </summary>
+        static readonly string[] textureNames = {
+            MeshSyncConstants._MetallicGlossMap,
+            MeshSyncConstants._BaseMap,
+            MeshSyncConstants._MaskMap
+        };
+
+        private List<Tuple<Material, string>> pendingMaterialUpdates = new();
+
+        /// <summary>
+        /// Returns the path where the given texture should be saved.
+        /// </summary>
+        /// <param name="mat">Material that owns the texture</param>
+        /// <param name="textureName">Name of the texture in the material</param>
+        /// <returns></returns>
+        private string GetSavePath(Material mat, string textureName) {
+            return Path.Combine(GetAssetsFolder(), $"{mat.name}_{textureName}_baked.png");
+        }
+
+        /// <summary>
+        /// Saves any render targets to the AssetDatabase as textures.
+        /// </summary>
+        private void SaveMaterialRenderTexturesToAssetDatabase() {
+            // Check if there are render textures that need to be saved to the asset database:
+            foreach (MaterialHolder materialHolder in materialList) {
+                foreach (string textureName in textureNames) {
+                    var mat = materialHolder.material;
+                    if (mat == null || !mat.HasTexture(textureName)) {
+                        continue;
+                    }
+
+                    var tex = mat.GetTexture(textureName);
+
+                    if (tex is RenderTexture renderTarget) {
+                        var texture = new Texture2D(renderTarget.width, renderTarget.height,
```

Comment:
I've added code to release it when they're saved as texture2d in SaveMaterialRenderTexturesToAssetDatabase and no longer used.
I don't think we can wrap them in a disposable class because they are only set on the material and we don't control its lifecycle except replacing them with textures saved to the asset database. We have no control over the user replacing them with their own texture or removing them from the material.